enhancements/network/vrf-cni.md

```diff
@@ -0,0 +1,161 @@
+
+The proposal is to introduce an additional CNI plugin that takes the vrf name as an input, and can be chained to the “main” cni used for the secondary interface, and that has the effect of creating the requested vrf (if it does not exist) and to add the interface to that vrf.
+
+In this way, the network interface returned by any CNI can be isolated in its own VRF, satisfying the above requirements without the need of additional capabilities.
+
+### User Stories [optional]
+
+#### Story 1
+
+As a user, I want to be able to optionally specify a VRF which the secondary nic is attached to.
+
+### Implementation Details/Notes/Constraints
+
+By adding several networks with the same `vrf` name, the CNI will ensure that they are all added to the same vrf, creating it only the first time.
+
+A `NetworkAttachmentDefinition` for MacVlan would look like:
+
+```yaml
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: macvlan-vrf-conf
+spec:
+  config: '{
+              "cniVersion": "0.3.1",
+              "plugins": [
+                {
+                  "type": "macvlan",
+                  "mode": "bridge",
+                  "ipam": {
+                    "type": "host-local",
+                    "subnet": "192.168.1.0/24",
+                    "rangeStart": "192.168.1.200",
+                    "rangeEnd": "192.168.1.216",
+                  }
+                },
+                {
+                  "type": "vrf",
+                  "vrfname": "blue",
+                  "routingtable": 1001
+                }
+              ]
+            }'
+```
+
+The same example for a SR-IOV device would look like:
+
+```yaml
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: sriov-vrf-conf
+  annotations:
+    k8s.v1.cni.cncf.io/resourceName: openshift.io/resource_id
+spec:
+  config: '{
+              "cniVersion": "0.3.1",
+              "plugins": [
+                {
+                  "type": "sriov",
+                  "ipam": {
+                    "type": "host-local",
+                    "subnet": "192.168.1.0/24",
+                    "rangeStart": "192.168.1.200",
+                    "rangeEnd": "192.168.1.216",
+                  }
+                },
+                {
+                  "type": "vrf",
+                  "vrfname": "blue",
+                  "routingtable": 1001
+                }
+              ]
+            }'
+```
+
+This can be supported at `NetworkAttachmentDefinition` level, or by documenting (and validating) chained CNIs in the `Network` CRD of `operator.openshift.io/v1`.
+
+At the same time, the `SriovNetwork` type from the SR-IOV operator needs to be extended for chaining meta CNIs.
+
+The CNIs we intend to support are SR-IOV and MacVlan.
+
+### Risks and Mitigations
+
+## Design Details
+
+This would be shipped as a new CNI plugin, ideally being part of the container networking plugins [repo](https://github.com/containernetworking/plugins),
```

Comment:
Has this made it onto the CNI maintainer's agenda yet?

Reply:
Nope, wasn't aware of that. Can you provide me a reference for that? (@s1061123 / @zshi-redhat maybe you also know?)

Reply:
I have no idea what's the agenda, Doug may provide the reference.